src/main/java/org/openjdk/jextract/JextractTool.java

```diff
@@ -117,26 +121,64 @@ public static Declaration.Scoped parse(List<Path> headers, String... parserOptio
 
     public static List<JavaSourceFile> generate(Declaration.Scoped decl, String headerName,
                                                 String targetPkg, List<String> libNames, PrintWriter errStream) {
-        return List.of(CodeGenerator.generate(decl, headerName, targetPkg, new IncludeHelper(), libNames, errStream));
+        return List.of(generate(decl, headerName, targetPkg, new IncludeHelper(), libNames, errStream));
     }
 
     private static List<JavaSourceFile> generateInternal(Declaration.Scoped decl, String headerName,
                                                          String targetPkg, IncludeHelper includeHelper,
                                                          List<String> libNames, PrintWriter errStream) {
-        return List.of(CodeGenerator.generate(decl, headerName, targetPkg, includeHelper, libNames, errStream));
+        return List.of(generate(decl, headerName, targetPkg, includeHelper, libNames, errStream));
+    }
+
+    private static JavaSourceFile[] generate(Declaration.Scoped decl, String headerName,
```

Comment:
Thanks - I think seeing everything here helps a bit - as opposed to having to chase multiple classes around which only contained a single method.